### PR TITLE
feat: add big lake connection

### PR DESCRIPTION
**Pull request by thadchais**
## Pull Request Submission Checklist

Please confirm that you have done the following before requesting reviews:

- [x] I have confirmed that the PR type is appropriate for the change I am making according to the [Honest Pull Request and Commit Message Naming Conventions](https://www.notion.so/honestbank/Pull-Request-and-Commit-Message-Naming-Conventions-bd97f2cbb34c4c73b1ff3a3e384b850c).
- [x] I have typed an adequate description that explains **why** I am making this change.
- [x] I have installed and run standard pre-commit hooks that lints and validates my code.

### Description

* 
Create Bigquery Connection Module to support BigLake table and other features

- - -
This change is [Reviewable](https://reviewable.io/reviews/honestbank/terraform-gcp-bigquery/24)

### CODE REVIEW

File: modules/gcp_bigquery_big_lake_connection/providers.tf [hunk 1]
```diff
@@ -0,0 +1,8 @@
+terraform {
+  required_providers {
+    google-beta = {
+      source  = "hashicorp/google"
+      version = ">=4.62.0"
```
Comment (MXfive) [inline modules/gcp_bigquery_big_lake_connection/providers.tf:5]:
```suggestion
      version = ">= 4.62.0"
```

File: modules/gcp_bigquery_big_lake_connection/outputs.tf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+output "service_account_id" {
+  description = "The connection name of the instance to be used in connection strings. For example, when connecting with Cloud SQL Proxy."
```
Comment (MXfive) [inline modules/gcp_bigquery_big_lake_connection/outputs.tf:2]: The description does not match the output name or value.

File: modules/gcp_bigquery_big_lake_connection/main.tf [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+# ######################################################################
+
+resource "google_bigquery_connection" "big_lake_connection" {
+  connection_id = var.connection_id
+  description   = var.description
+  friendly_name = var.connection_id
+  location      = var.location
+  provider      = google
+
+  cloud_resource {}
```
Comment (MXfive) [inline modules/gcp_bigquery_big_lake_connection/main.tf:12]: This is optional and not used, should be deleted unless somehow is needed?

**Review by MXfive — CHANGES_REQUESTED** [review]
Requested a couple of changes, but I don't understand the code really, can we do a quick sync review please.

File: modules/gcp_bigquery_big_lake_connection/main.tf (hunk 3 above)
Comment (thadchais) [reply modules/gcp_bigquery_big_lake_connection/main.tf:12]: There are many types of connections `cloud_resource` must be defined to use the Big Lake table.
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/bigquery_connection#example-usage---bigquery-connection-cloud-resource

Comment (thadchais) [reply modules/gcp_bigquery_big_lake_connection/main.tf:12]: image

**Review by MXfive — APPROVED** [review]

**Review by xoxwaw — APPROVED** [review]

**Pull request merged**